src/org.hdfgroup.hdfview/hdf/view/TableView/DataProviderFactory.java

```diff
@@ -1661,11 +1662,11 @@ public Object getDataValue(int columnIndex, int rowIndex) {
                     rElements = (byte[]) dataBuf;
 
                 if (h5dtype.isStdRef())
-                    theValue = populateReference(rElements, bufIndex);
```

Comment:
Hmm these changes don't cause problems by not taking the bufIndex into account at all?

Reply:
References are handled differently - there might be a better way, but it wasn't obvious.

Reply:
Is it because the `dataBuf` here is an instance of `ArrayList`? The code above takes `bufIndex` into account when retrieving from an `ArrayList`, but not when `dataBuf` is just a `byte[]`. Seems like this change could work for one case and break the other.

Reply:
Right at this point with the release, I didn't want to cause a major disruption. There is probably a lot more cleanup that could happen. But everything passes and we need to to get some fixes out the door.

Reply:
This looks more like something needs to actually be fixed for functionality reasons rather than just cleanup, but I supposed we'll see if some other reference case is broken after this.